backend/src/controllers/task.ts

```diff
@@ -26,12 +27,27 @@ export const getTask: RequestHandler = async (req, res, next) => {
 };
 
 export const createTask: RequestHandler = async (req, res, next) => {
-  const { title, description } = req.body;
+  // fetch errors returned by the validator
+  const errors = validationResult(req);
+  const { title, description, isChecked } = req.body;
 
   try {
+    if (!errors.isEmpty()) {
+      let errorString = "";
+
+      // parse through errors returned by the validator and append them to the error string
+      for (const error of errors.array()) {
+        errorString += error.msg + " ";
+      }
+
+      // trim removes the trailing space created in the for loop
+      throw createHttpError(400, errorString.trim());
+    }
```

Comment:
I think we can pull the error message generation into a helper function which can throw errors and move more stuff into the try block like this:

```javascript
try {
  checkValidationResult(req);
  const { title, description, isChecked } = req.body;
  const task = await TaskModel.create({ /* ... */ });
  res.status(201).json(task);
}
```

Reply:
I've been keeping `const { title, description, isChecked } = req.body;` out of the try because it doesn't throw an error.

Reply:
Good call lol